### PR TITLE
Avoid using QUIT command

### PR DESCRIPTION
**Pull request by sazzad16**
Backport of https://github.com/redis/jedis/pull/3353 

---------

Please refer to: https://github.com/redis/redis/issues/11420

This PR cancels calling the quit command when the Jedis internal link is abnormal, and deprecate the user interface `Jedis.quit()`. This optimizes Jedis connection pool performance and fixes: https://github.com/redis/jedis/issues/2105

### CODE REVIEW

File: src/main/java/redis/clients/jedis/BinaryJedis.java [hunk 1]
```diff
@@ -110,9 +110,6 @@ private void initializeFromClientConfig(JedisClientConfig config) {
       }
     } catch (JedisException je) {
       try {
-        if (isConnected()) {
-          quit();
-        }
         disconnect();
```
Comment (yangbodong22011) [inline src/main/java/redis/clients/jedis/BinaryJedis.java:113]:
```suggestion
        setBroken();
        disconnect();
```
Jedis4 actively called `setBroken()` in the `quit()` command, we can also add it here.

**Comment by yangbodong22011** [comment]
The other code LGTM.

File: src/main/java/redis/clients/jedis/BinaryJedis.java (hunk 1 above)
Comment (sazzad16) [reply src/main/java/redis/clients/jedis/BinaryJedis.java:113]: @yangbodong22011 We don't have `setBroken()` in Jedis 3.x.

Comment (yangbodong22011) [reply src/main/java/redis/clients/jedis/BinaryJedis.java:113]: ok, no problem then.

**Comment by rogoza-igor** [comment]
Hello! Thank you for backport. But could you explain please why did you revert changes for sharded classes?

**Pull request merged**